### PR TITLE
add vulnerability case

### PR DESCRIPTION
**Pull request by sigsbee**

### CODE REVIEW

File: tests/testcases/Group10-Vulnerability/10-01-Cliar-can-not-scan-some-images.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# References:    
+User guide  
+  
+# Environment:  
+* This test requires that a Harbor instance is running and available.  
+* Harbor is installed with clair enable.  
+* A linux host with Docker CLI installed.  
+  
+# Test Steps:  
+1. Login Harbor.
```
Comment (reasonerjt) [inline tests/testcases/Group10-Vulnerability/10-01-Cliar-can-not-scan-some-images.md]: A prepreq is Clair's Vuln DB has updated successfully.

File: tests/testcases/Group10-Vulnerability/10-01-Cliar-can-not-scan-some-images.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  
+# Test Steps:  
+1. Login Harbor.  
+2. Push an image clair does not recognize(for example: photon).  
+3. Scan the pushed image.  
+4. View the scan result.  
+  
+# Expected Outcome:  
+* Step3 should show view log link.  
+* Step4 will show scan log
```
Comment (reasonerjt) [inline tests/testcases/Group10-Vulnerability/10-01-Cliar-can-not-scan-some-images.md]: Verify the content of the log please.

File: tests/testcases/Group10-Vulnerability/10-01-Cliar-can-not-scan-some-images.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+# Test Steps:  
+1. Login Harbor.  
+2. Push an image clair does not recognize(for example: photon).  
+3. Scan the pushed image.  
+4. View the scan result.  
+  
+# Expected Outcome:  
+* Step3 should show view log link.  
+* Step4 will show scan log  
+
```
Comment (reasonerjt) [inline tests/testcases/Group10-Vulnerability/10-01-Cliar-can-not-scan-some-images.md]: Should verify whether it's possible to rescan.

File: tests/testcases/Group10-Vulnerability/10-02-Clair-scan-a-image-with-vulnerability.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+To verify that clair can scan a image with vulnerability.  
+
+# References:  
+User guide  
+  
+# Environment:  
+* This test requires that a Harbor instance is running and available.  
+* Harbor is installed with clair enable.  
+* A linux host with Docker CLI installed.  
+
```
Comment (reasonerjt) [inline tests/testcases/Group10-Vulnerability/10-02-Clair-scan-a-image-with-vulnerability.md]: Again, an important prereq is:
A prepreq is Clair's Vuln DB has updated successfully.

File: tests/testcases/Group10-Vulnerability/10-02-Clair-scan-a-image-with-vulnerability.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+User guide  
+  
+# Environment:  
+* This test requires that a Harbor instance is running and available.  
+* Harbor is installed with clair enable.  
+* A linux host with Docker CLI installed.  
+  
+# Test Step  
+1. Login Harbor.  
+2. Push a normal image.
```
Comment (reasonerjt) [inline tests/testcases/Group10-Vulnerability/10-02-Clair-scan-a-image-with-vulnerability.md]: What is normal image? Please be specific.

File: tests/testcases/Group10-Vulnerability/10-02-Clair-scan-a-image-with-vulnerability.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  
+# Test Step  
+1. Login Harbor.  
+2. Push a normal image.  
+3. Scan the image.  
+4. Click tag link enter detail page.  
+  
+# Expected Outcome:  
+* Step3 should exist result summary chart.  
+* Step4 should see cve link.
```
Comment (reasonerjt) [inline tests/testcases/Group10-Vulnerability/10-02-Clair-scan-a-image-with-vulnerability.md]: Should verify:
1)The summary matches between the summary and details.
2)CVE link is clickable and point to the correct location on the internet.
3)Try to rescan, and what happens?
4)Does the color bar reflect the correct ratio of the vulnerabilities?

Some error case, shut down registry and scan, what will happen? What logs should be seen?
start registry and rescan what will happen?

File: tests/testcases/Group10-Vulnerability/10-03-Clair-scan-0-package-image.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+# Purpose:  
+To verify clair scan result of a 0 package image.  
+  
+# References:  
+User guide  
+  
+# Environment:  
+* This test requires that a Harbor instance is running and available.  
+* Harbor is installed with clair enable.  
+* A linux host with Docker CLI installed.
```
Comment (reasonerjt) [inline tests/testcases/Group10-Vulnerability/10-03-Clair-scan-0-package-image.md]: Again, an important prereq is:
A prepreq is Clair's Vuln DB has updated successfully.

File: tests/testcases/Group10-Vulnerability/10-03-Clair-scan-0-package-image.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  
+# Environment:  
+* This test requires that a Harbor instance is running and available.  
+* Harbor is installed with clair enable.  
+* A linux host with Docker CLI installed.  
+  
+# Test Step  
+1. Login Harbor.  
+2. Push a helloworld or busybox image.  
+3. Scan image.
```
Comment (reasonerjt) [inline tests/testcases/Group10-Vulnerability/10-03-Clair-scan-0-package-image.md]: I think in some case you should add steps to verify only project admin can scan the image.

File: tests/testcases/Group10-Vulnerability/10-04-Clair-data-not-ready-hint.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+# Purpose:  
+To verify there will be a hint if vulnerability data is not ready.  
+  
+# References:  
+User guide  
+  
+# Environment:  
+* This test requires that a Harbor instance is running and available.  
+* Harbor is installed with clair enable.  
+* A linux host with Docker CLI installed.
```
Comment (reasonerjt) [inline tests/testcases/Group10-Vulnerability/10-04-Clair-data-not-ready-hint.md]: Make sure the env has poor or no internet connection, so the DB won't be ready.

File: tests/testcases/Group10-Vulnerability/10-05-Schedule-daily-scan.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+User guide  
+  
+# Environment:  
+* This test requires that a Harbor instance is running and available.  
+* Harbor is installed with clair enable.  
+* A linux host with Docker CLI installed.  
+  
+# Test Step  
+1. Login Harbor.  
+2. Push an image.
```
Comment (reasonerjt) [inline tests/testcases/Group10-Vulnerability/10-05-Schedule-daily-scan.md]: You have to push more than one images to more than one project

File: tests/testcases/Group10-Vulnerability/10-05-Schedule-daily-scan.md [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+# Environment:  
+* This test requires that a Harbor instance is running and available.  
+* Harbor is installed with clair enable.  
+* A linux host with Docker CLI installed.  
+  
+# Test Step  
+1. Login Harbor.  
+2. Push an image.  
+3. Schedule a daily scan.  
+4. Wait one day.
```
Comment (reasonerjt) [inline tests/testcases/Group10-Vulnerability/10-05-Schedule-daily-scan.md]: 1. Is it possible to schedule some time before the current time?
2. If schedule a scan to 1 hour later, why wait for one day?

File: tests/testcases/Group10-Vulnerability/10-06-Change-scan-configuration.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+1. Login harbor as admin.   
+2. Push a image to a project.   
+3. Click configuration.   
+4. Click vunlerablity.   
+5. Set scan all to none.   
+6. Click scan now.   
+7. Check the image scan time.   
+   
+# Expected Outcome:   
+* Step 4 if a new installed Harbor,vulnerabilit not ready hit should exist.
```
Comment (reasonerjt) [inline tests/testcases/Group10-Vulnerability/10-06-Change-scan-configuration.md]: fix typo.

File: tests/testcases/Group10-Vulnerability/10-05-Schedule-daily-scan.md [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+* Harbor is installed with clair enable.  
+* A linux host with Docker CLI installed.  
+  
+# Test Step  
+1. Login Harbor.  
+2. Push an image.  
+3. Schedule a daily scan.  
+4. Wait one day.  
+5. Check the timestamp of scan .  
+
```
Comment (reasonerjt) [inline tests/testcases/Group10-Vulnerability/10-05-Schedule-daily-scan.md]: should also test the case when the daily scan is disabled.  Maybe in a separate case and think about how to verify it quickly without having to wait for one day.

File: tests/testcases/Group10-Vulnerability/10-06-Change-scan-configuration.md [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+2. Push a image to a project.   
+3. Click configuration.   
+4. Click vunlerablity.   
+5. Set scan all to none.   
+6. Click scan now.   
+7. Check the image scan time.   
+   
+# Expected Outcome:   
+* Step 4 if a new installed Harbor,vulnerabilit not ready hit should exist.  
+* Step 5 schedule time input should disappear.
```
Comment (reasonerjt) [inline tests/testcases/Group10-Vulnerability/10-06-Change-scan-configuration.md]: Should also verify the scan now is disabled, right?

Also after that period of time, he should be able to trigger "scan now" again.

**Comment by reasonerjt** [comment]
Please also add some longevity test as there's updater running every a few hours.

Also,  could you add test case when there's no internet connection, where user has to import data manually do some scan, and keep it running for a few days?

File: tests/testcases/Group10-Vulnerability/10-03-Clair-scan-0-package-image.md [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+* Clair has been updated to the latest.    
+
+# Test Step  
+1. Login Harbor as admin.  
+2. Create a project.  
+3. Push a helloworld or busybox image to the created project.  
+4. Scan the image.  
+  
+# Expected Outcome:  
+* Step 4 result should be 0 of 0.
```
Comment (steven-zou) [inline tests/testcases/Group10-Vulnerability/10-03-Clair-scan-0-package-image.md]: Severity : Unknown

**Pull request merged**